internal/secrets/secret_test.go

```diff
@@ -74,77 +65,24 @@ func TestFetchSecrets_Success(t *testing.T) {
 func TestFetchSecrets_EmptyKeys(t *testing.T) {
 	t.Parallel()
 
-	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		switch req.URL.Path {
-		case "/jobs/test-job-id/secrets":
-			switch req.URL.Query()["key"][0] {
-			case "DATABASE_URL":
-				rw.WriteHeader(http.StatusOK)
-				fmt.Fprintf(rw, `{"key": "DATABASE_URL", "value": "postgres://user:pass@host:5432/db"}`)
-			case "API_TOKEN":
-				rw.WriteHeader(http.StatusOK)
-				fmt.Fprintf(rw, `{"key": "DATABASE_URL", "value": "secret-token-123"}`)
-			case "MISSING":
-				rw.WriteHeader(http.StatusNotFound)
-				fmt.Fprintf(rw, `{"message": "secret not found"}`)
-			default:
-				rw.WriteHeader(http.StatusNotFound)
-				fmt.Fprintf(rw, `{"message": "secret not found"}`)
-			}
-		default:
-			t.Errorf("Unknown endpoint %s %s", req.Method, req.URL.Path)
-			http.Error(rw, "Not found", http.StatusNotFound)
-		}
-	}))
-
-	apiClient := api.NewClient(logger.Discard, api.Config{
-		Endpoint: server.URL,
-		Token:    "llamas",
-	})
-
-	secrets, errs := FetchSecrets(context.Background(), apiClient, "test-job-id", []string{}, 10)
+	apiClient := api.NewClient(logger.Discard, api.Config{Token: "llamas"})
```

Comment:
This client would send requests to the default (real) API, is that intentional? Shouldn't there still be a test server?

Reply:
my thought was that because it short-circuits and just returns an empty slice when the  `keys` slice is empty, it doesn't need one, but that's testing internal behaviour implicitly. lemme re-add something to make it clear it doesn't hit the endpoint